src/katgpucbf/xbgpu/engine.py

```diff
@@ -528,254 +920,45 @@ async def _receiver_loop(self) -> None:
-                    tx_item = await self._flush_accumulation(tx_item, current_accum)
-                self.correlation.last_batch = i + 1
-                current_timestamp += self.rx_heap_timestamp_step
-
-            do_correlation()
-            # If the last batch of the chunk was also the last batch of the
-            # accumulation, we can flush it now without waiting for more data.
-            # This is mostly a convenience for unit tests, since in practice
-            # we'd expect to see more data soon.
-            current_accum = current_timestamp // self.timestamp_increment_per_accumulation
-            tx_accum = tx_item.timestamp // self.timestamp_increment_per_accumulation
-            if current_accum != tx_accum:
-                tx_item = await self._flush_accumulation(tx_item, current_accum)
+        for pipeline in self._pipelines:
+            pipeline.shutdown()
 
-            rx_item.reset()
-            rx_item.add_marker(self._proc_command_queue)
-            await self._rx_free_item_queue.put(rx_item)
+    def _request_pipeline(self, stream_name: str) -> Pipeline:
+        """Get the pipeline related to the katcp request.
 
-        # When the stream is closed, if the sender loop is waiting for a tx item,
-        # it will never exit. Upon receiving this NoneType, the sender_loop can
-        # stop waiting and exit.
-        logger.debug("_gpu_proc_loop completed")
-        self._tx_item_queue.put_nowait(None)
+        Return the first Pipeline that matches by name.
 
-    async def _sender_loop(self) -> None:
+        Raises
+        ------
+        FailReply
+            If the `stream_name` is not a known output.
         """
-        Send heaps to the network in a continuous loop.
+        for pipeline in self._pipelines:
+            if stream_name == pipeline.output.name:
+                return pipeline
+        raise aiokatcp.FailReply(f"No output stream called {stream_name!r}")
 
-        This function does the following:
-        1. Get an item from the _tx_item_queue.
-        2. Wait for all the events on this item to complete.
-        3. Wait for an available heap buffer from the send_stream.
-        4. Transfer the GPU buffer in the item to the heap buffer in system RAM.
-        5. Wait for the transfer to complete.
-        6. Transmit data in heap buffer out into the network.
-        7. Place the tx_item on _tx_item_free_queue so that it can be reused.
+    async def request_capture_start(self, ctx, stream_name: str) -> None:
+        """Start transmission of a stream.
 
-        The above steps are performed in a loop until the running flag is set to false.
-
-        NOTE: The transfer from the GPU to the heap buffer and the sending onto
-        the network could be pipeline a bit better, but this is not really
-        required in this loop as this whole process occurs at a much slower
-        pace than the rest of the pipeline.
+        The request requires a stream name to enable
```

Comment:
The generated help for katcp requests doesn't show the function signature, so it's recommended to use a `Parameters` section to formally document the parameter.

Reply:
Rather embarassingly, I didn't know you could `?help <request_name>` like you can with ipython `thing.function?`.

Updating now!